src/odemis/gui/util/img.py

```diff
@@ -2098,3 +2099,107 @@ def wxImageScaleKeepRatio(im, size, quality=wx.IMAGE_QUALITY_NORMAL):
+    corresponding part of the ovv and the previous content at this part of the ovv is
+    deleted. If the tile reaches beyond the borders of the ovv, it is cropped.
+    tile: 3D DataArray (RGB or RGBA) with MD_PIXEL_SIZE and MD_POS metadata
+    ovv: 3D DataArray (RGB or RGBA) with MD_PIXEL_SIZE metadata
+    Returns 3D DataArray with the same shape as tile (updated ovv)
+    """
+    # TODO: use cairo to insert images. Not required for current specification, but
+    # 'blend_screen' mode insertion might be necessary to overlay multiple optical images
+    # in future implementations.
+
+    # Tile parameters
+    tile_pos_m = tile.metadata[model.MD_POS]
+    tile_mpp = tile.metadata[model.MD_PIXEL_SIZE]
+    x, y, c = tile.shape
+    tile_size_m = (x * tile_mpp[0], y * tile_mpp[1])
+
+    # Ovv parameters
+    ovv_mpp = ovv.metadata[model.MD_PIXEL_SIZE]
+    ovv_sz = ovv.shape[1], ovv.shape[0]
+
+    # Tile size, pos in ovv image (px)
+    tile_sz_px = (int(tile_size_m[0] / ovv_mpp[0]), int(tile_size_m[1] / ovv_mpp[1]), c)
+    tile_pos_px = (int(tile_pos_m[0] / ovv_mpp[0]), int(tile_pos_m[1] / ovv_mpp[1]))
+    # Change origin of coordinates from center of the image to top left of ovv image,
+    # the position indicates the top left part of the tile, not its center.
+    pos_top_left = (int(tile_pos_px[0] + ovv_sz[0] / 2 - tile_sz_px[0] / 2),
+                    int(ovv_sz[1] / 2 - tile_pos_px[1] - tile_sz_px[1] / 2))
+
+    # Crop tile on edges of overview image
+    if pos_top_left[0] + tile_sz_px[0] > ovv_sz[0]:
+        diff_x2 = pos_top_left[0] + tile_sz_px[0] - ovv_sz[0]
+        x_right = tile_sz_px[0] - diff_x2
+    else:
+        diff_x2 = 0
+        x_right = tile_sz_px[0]
+
+    if pos_top_left[0] < 0:
+        diff_x = -pos_top_left[0]
+        pos_top_left = (0, pos_top_left[1])
+    else:
+        diff_x = 0
+
+    if pos_top_left[1] + tile_sz_px[1] > ovv_sz[1]:
+        diff_y2 = pos_top_left[1] + tile_sz_px[1] - ovv_sz[1]
+        y_right = tile_sz_px[1] - diff_y2
+    else:
+        diff_y2 = 0
+        y_right = tile_sz_px[1]
+
+    if pos_top_left[1] < 0:
+        diff_y = -pos_top_left[1]
+        pos_top_left = (pos_top_left[0], 0)
+    else:
+        diff_y = 0
+
+    ovv[pos_top_left[1]: pos_top_left[1] + tile_sz_px[1] - diff_y - diff_y2,
+        pos_top_left[0]: pos_top_left[0] + tile_sz_px[0] - diff_x - diff_x2] = \
+        img.rescale_hq(tile, (tile_sz_px[1], tile_sz_px[0], c))[diff_y:y_right, diff_x:x_right]
+    return ovv
+
+def merge_screen(im, background):
+    """ 
+    Merges two images (im and background) into one.
+    im, background: DataArray with opt.shape = background.shape (either RGB or RGBA)
+    returns RGB DataArray
+    """
+    assert(im.shape == background.shape, "Images have different shapes.")
+    if im.shape[-1] != 3 and im.shape[-1] != 4:
+        raise ValueError("Ovv images have an invalid number of channels: %d" % (im.shape[-1]))
+
+    md1 = copy.deepcopy(im.metadata)
```

Comment:
You are now not using `md1` and `md2` at all. You need to also change im into a new DataArray with that new metadata. You can do it with this line:
`im = model.DataArray(im, md1)`

BTW, it seems that `dc_keepalpha` is not useful for `background`, so you could avoid duplicating the metadata.